stan/math/rev/core/precomputed_gradients.hpp

```diff
@@ -87,18 +190,29 @@ class precomputed_gradients_vari : public vari {
  * @tparam Arith An arithmetic type
  * @tparam VecVar A vector of vars
  * @tparam VecArith A vector of arithmetic types
+ * @tparam ContainerOperands tuple of any container operands (var_value
+ * containing Eigen types)
+ * @tparam ContainerGradients tupleof any container gradients (Eigen types)
  * @param[in] value The value of the resulting dependent variable.
  * @param[in] operands operands.
  * @param[in] gradients vector of partial derivatives of result with
  * respect to operands.
+ * @param container_operands any container operands
+ * @param container_gradients any container gradients
  * @return An autodiff variable that uses the precomputed
  * gradients provided.
  */
-template <typename Arith, typename VecVar, typename VecArith>
-inline var precomputed_gradients(Arith value, VecVar&& operands,
-                                 VecArith&& gradients) {
-  return {new precomputed_gradients_vari(value, std::forward<VecVar>(operands),
-                                         std::forward<VecArith>(gradients))};
+template <typename Arith, typename VecVar, typename VecArith,
+          typename... ContainerOperands, typename... ContainerGradients>
+inline var precomputed_gradients(
+    Arith value, const VecVar& operands, const VecArith& gradients,
+    const std::tuple<ContainerOperands...>& container_operands = std::tuple<>(),
+    const std::tuple<ContainerGradients...>& container_gradients
+    = std::tuple<>()) {
+  return {new precomputed_gradients_vari_template<
```

Comment:
So everything will be copied to the stack on construction here?

Reply:
I am modifying this a bit. Now it is just everything not already on stack.